### PR TITLE
Connect endpoints to frontend

### LINKED ISSUE #22: [TP][BE] Endpoint Integration for CRUD events
Story Points: 2

### PR DESCRIPTION
**Pull request by shizuko-akamoto**
Hmm... guess I lied when I said I'm breaking it down. Somehow this one turned out to be another gigantic PR again. I'm sorry guys 😢  Rather than looking at each changes, maybe run the app and confirm if the behaviours are as expected.

Most of the changes are coming from the fact that I had to change the client side to use `Destination` and `Trecipe` interfaces in the shared folder.

- Basically added bunch of service methods for calling server endpoints, and connected them to each page.
- Also fixed one bug with destinations getting wiped out on reload on Trecipe and Map page (#74)

To come:
- 3XX, 5XX error handling on client side
- extracting photos from Place Details request
- photo uploading for trecipe (this one is done, only need to make a PR on top of this)

Closes #22

### CODE REVIEW

File: packages/client/src/components/Map/mapHelper.tsx [hunk 1]
```diff
@@ -70,31 +70,36 @@ export class AutoComplete {
...
 };
 
 // Get destination category based on the types return by the google's place API
-export const getDestCategory = (types: Array<string> | undefined) => {
+export const getDestCategory = (types: Array<string> | undefined): Array<DestinationCategory> => {
     if (types) {
         for (const [key, value] of Object.entries(destTypes)) {
             if (intersection(types, value).length > 0) {
-                return key;
+                return [key as DestinationCategory];
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply packages/client/src/components/Map/mapHelper.tsx:98]: Lol you're right. Clearly was zoned out when writing this, lemme rewrite this forloop.

File: packages/client/src/redux/util.ts [hunk 2]
```diff
@@ -10,3 +14,10 @@ export function typedAction<T extends string, P extends any>(
 export function typedAction(type: string, payload?: any) {
     return { type, payload };
 }
+
+export type AppThunk<ReturnType = void> = ThunkAction<
+    ReturnType,
+    RootState,
+    unknown,
+    Action<string>
+>;
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply packages/client/src/redux/util.ts:23]: DW. I HAVE NO IDEA TOO

File: packages/client/src/pages/Map/DestinationCard/DestinationCard.tsx [hunk 3]
```diff
@@ -59,14 +61,18 @@ export class DestinationCard extends React.Component<DCProps> {
...
                             </h6>
-                            <h6 id="dest-location">{destModel.address}</h6>
+                            <h6 id="dest-location">{destModel.formattedAddress}</h6>
                         </span>
                         <RatingBar rating={destModel.rating} />
                     </div>
-                    <p id="dest-description">{destModel.description}</p>
+                    <p id="dest-description">
+                        {destModel.website}
+                        {destModel.formattedPhoneNumber}
```
[the remark this answers is not in the document]
Comment (shizuko-akamoto) [reply packages/client/src/pages/Map/DestinationCard/DestinationCard.tsx:74]: Ohhh it does indeed.....and no that was not intended. Will fix!

File: packages/client/src/redux/util.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (knox153) [reply packages/client/src/redux/util.ts:23]: LMAO

**Pull request merged**